ldap-api/src/main/java/ca/bc/gov/hlth/ldapapi/service/LdapService.java

```diff
@@ -70,43 +72,33 @@ public User authenticate(UserCredentials userCredentials) {
         return createReturnMessage(userInfo.getName(), validCredentials, userUnlocked, lockoutTimeInHours, remainingAttempts, userInfo.getAttributes());
     }
 
-    private SearchResult searchUser(String username) {
-        try {
-            DirContext ctx = new InitialDirContext(ldapProperties);
-            SearchControls constraints = new SearchControls();
-            constraints.setSearchScope(2);
-            constraints.setCountLimit(300L); // TODO filter out names that could cause a wildcard search
-            constraints.setReturningObjFlag(false);
-            NamingEnumeration<SearchResult> results = ctx.search(LDAP_SEARCH_BASE, "uid=" + username, constraints);
-            ctx.close();
-            if (results.hasMore()) {
-                return results.next();
-            } else {
-                return null;
-            }
-
-
-        } catch (NamingException e) {
-            throw new RuntimeException(e);
+    private SearchResult searchUser(String username) throws NamingException {
+        DirContext ctx = new InitialDirContext(ldapProperties);
+        SearchControls constraints = new SearchControls();
+        constraints.setSearchScope(2);
+        constraints.setCountLimit(300L); // TODO filter out names that could cause a wildcard search
+        constraints.setReturningObjFlag(false);
+        NamingEnumeration<SearchResult> results = ctx.search(LDAP_SEARCH_BASE, "uid=" + username, constraints);
+        ctx.close();
+        if (results.hasMore()) {
+            return results.next();
+        } else {
+            return null;
         }
```

Comment:
Any uncaught exception will result in returning an HTTP 500 error to the client, which is what we want. The stack trace will be automatically logged.